internal/ethapi/api.go

```diff
@@ -1998,6 +1998,51 @@ func (s *PublicNetAPI) Version() string {
 	return fmt.Sprintf("%d", s.networkVersion)
 }
 
+// PublicDbAPI exposes low-level methods for interacting with the database.
+type PublicDbAPI struct {
+	b Backend
+}
+
+// NewPublicDbAPI creates a new instance of PublicDbAPI.
+func NewPublicDbAPI(b Backend) *PublicDbAPI {
+	return &PublicDbAPI{b: b}
+}
+
+// Get returns the raw value of a key stored in the database.
+func (api *PublicDbAPI) Get(key string) (hexutil.Bytes, error) {
+	blob, err := parseHexOrString(key)
+	if err != nil {
+		return nil, err
+	}
+	return api.b.ChainDb().Get(blob)
+}
+
+// Put stores a raw value under the key in the database.
+func (api *PublicDbAPI) Put(key string, value hexutil.Bytes) (hexutil.Bytes, error) {
+	var (
+		ret []byte
+		db  = api.b.ChainDb()
+	)
+	blob, err := parseHexOrString(key)
+	if err != nil {
+		return nil, err
+	}
+	data, err := db.Get(blob)
+	if err == nil {
+		ret = data
+	}
+	return ret, db.Put(blob, value)
+}
+
+// Delete removes an item from the database.
+func (api *PublicDbAPI) Delete(key string) error {
```

Comment:
Should we return the previous value as well? So that users can see clearly what's deleted.